addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.onewiregpio.handler;
+
+import static org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OneWireGPIOHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class OneWireGPIOHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(OneWireGPIOHandler.class);
+
+    private String gpioBusFile;
+    private Integer refreshTime;
+
+    public OneWireGPIOHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(TEMPERATURE)) {
+            if (command instanceof RefreshType) {
+                publishSensorValue(channelUID);
+            } else {
+                logger.debug("Command {} is not supported for channel: {}. Supported command: REFRESH", command,
+                        channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            checkConfiguration();
+            updateStatus(ThingStatus.ONLINE);
+            startAutomaticRefresh();
+            super.initialize();
+
+        } catch (Exception e) {
+            logger.warn("Problem initializing thing: {}.", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
In case you want (and can, regarding the exception being caught) to be more specific about the details of going offline you can add `ThingStatusDetail` as 2nd parameter to the #updateStatus method.